server/svix-server/src/v1/utils.rs

```diff
@@ -63,6 +64,44 @@ impl<T: Validate> Validate for ReversibleIterator<T> {
     }
 }
 
+pub enum ReversibleSelect<T: QuerySelect> {
+    Forward(T),
+    Reversed(T),
+}
+
+pub fn apply_pagination<
```

Comment:
I'm not sure this actually makes things simpler. Between `ReversibleSelect` and and everything I think it just adds complexity rather than reduces it.

I can *maybe* see value in a function that accepts an iterator, after, before and returns ReversibleIterator that can be shared everywhere. Heck, I can even see value in the other part being a function (even called apply_pagination). Though I think passing before/after which are *not* pagination related into a pagination function is confusing. And no, changing the function name won't help. :)

What do you think?